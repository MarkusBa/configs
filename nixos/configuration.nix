# Edit this configuration file to define what should be installed on
# your system.  Help is available in the configuration.nix(5) man page
# and in the NixOS manual (accessible by running ‘nixos-help’).

{ config, pkgs, ... }:

{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
    ];

  # Use the GRUB 2 boot loader.
  boot.loader.grub.device = "/dev/sdb";
  boot.loader.grub.enable = true;
  boot.loader.grub.version = 2;
  # Define on which hard drive you want to install Grub.

  nix.extraOptions = ''
   binary-caches-parallel-connections = 1
  '';
  
  #haskell
  nix.binaryCaches = [ https://cache.nixos.org/ http://hydra.nixos.org/ http://hydra.cryp.to/ ];
  nix.trustedBinaryCaches = [ "http://hydra.nixos.org" "http://hydra.cryp.to" ];

  networking.hostName = "nixos"; # Define your hostname.
  networking.hostId = "a671cc0d";
  #networking.wireless.enable = true;  # Enables wireless.
  networking.networkmanager.enable = true;
  services.dbus.enable = true;  

  # Select internationalisation properties.
  i18n = {
    consoleFont = "lat9w-16";
    consoleKeyMap = "de";
    defaultLocale = "de_DE.UTF-8";
  };

  nixpkgs.config.allowUnfree = true;

  # test
  # List packages installed in system profile. To search by name, run:
  # $ nix-env -qaP | grep wget
  environment.systemPackages = with pkgs; [
    #haskellPackages.purescript    
    haskellPackages.yi
    xlibs.xev
    kde4.ktorrent
    #automake
    #gnumake
    #sbcl
    #ruby_2_2_0
    #chibi
    #leksah
    sqlite
    wireshark
    vimPlugins.vim-addon-syntax-checker
    vimPlugins.vimshell
    vimPlugins.idris-vim
    vimPlugins.pathogen
    gnome3.nautilus
    kde4.ksnapshot
    imagemagick
    youtube-dl
    maven
    idea.idea-community
    vlc
    #haskellPackages.regexBase
    #haskellPackages.regexPcre
    #haskellPackages.HTTP    
    #haskellPackages.regexPosix
    #haskellPackages.yesod
    #haskellPackages.text
    #haskellPackages.persistent
    #haskellPackages.persistentSqlite
    #haskellPackages.persistentPostgresql
    #haskellPackages.transformersBase
    #haskellPackages.resourcet
    #haskellPackages.monadLogger
    pwgen
    #eclipses.eclipse_sdk_431
    eclipses.eclipse_sdk_442
    #haskellPackages.purescript
    #haskellPackages.cabal2nix
    #haskellPackages.idris
    nodejs
    #opam
    #ocaml
    #ocamlPackages.ocaml_pcre
    #ocamlPackages.utop
    kde4.kdegraphics_thumbnailers
    #haskellPackages.ghc
    #haskellPackages.cabalInstall
    unzip
    gnum4
    kde4.okular
    kde4.ark
    lighttable
    #findutils
    #haskellPackages.xmonad
    #wpa_supplicant_gui
    vim
    wget
    chromium
    networkmanager
    networkmanagerapplet
    emacs
    xlibs.xmodmap
    firefox
    thunderbird
    gnupg
    zsh
    lynx
    #mysql
    #mysqlWorkbench
    oraclejdk8
    pgadmin  
    gitAndTools.gitFull
    postgresql94
    leiningen
    kde4.yakuake
    
  ];

  # List services that you want to enable:

  services.virtualboxHost.enable = true;
  users.extraGroups.vboxusers.members = [ "markus" ];  

  services.locate.enable = true;
  services.locate.period = "33 7 * * *";

  services.postgresql.enable = true;
  services.postgresql.package = pkgs.postgresql93;

  # Enable the OpenSSH daemon.
  # services.openssh.enable = true;

  # Enable CUPS to print documents.
  services.printing.enable = true;

  # Enable the X11 windowing system.
  services.xserver.enable = true;
  services.xserver.layout = "de";
  services.xserver.xkbOptions = "eurosign:e";

  #group networkmanager
  services.xserver.windowManager.icewm.enable = true;
  services.xserver.windowManager.xmonad.enable = true;     # installs xmonad and makes it available
  #services.xserver.windowManager.xmonad.enableContribAndExtras = true; # makes xmonad-contrib and xmonad-extras available
  #services.xserver.windowManager.default       = "xmonad"; # sets it as default
  #services.xserver.desktopManager.default      = "none";   # the plain xmonad experience
  services.xserver.windowManager.default = "icewm";

  # Enable the KDE Desktop Environment.
  services.xserver.displayManager.kdm.enable = true;
  services.xserver.desktopManager.kde4.enable = true;  
  
  #services.xserver.displayManager.gdm.enable = true;
  #services.xserver.desktopManager.xfce.enable = true;

  # Define a user account. Don't forget to set a password with ‘passwd’.
  users.extraUsers.guest = {
    isNormalUser = true;
    uid = 1000;
    name = "markus";
    group = "users";
    extraGroups = ["wheel" "networkmanager"];
    createHome = true;
    home = "/home/markus";
  };

}
