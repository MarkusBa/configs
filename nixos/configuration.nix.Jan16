# Edit this configuration file to define what should be installed on
# your system.  Help is available in the configuration.nix(5) man page
# and in the NixOS manual (accessible by running ‘nixos-help’).

{ config, pkgs, ... }:

{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
    ];

  # Use the GRUB 2 boot loader.
  boot.loader.grub.enable = true;
  boot.loader.grub.version = 2;
  # Define on which hard drive you want to install Grub.
  boot.loader.grub.device = "/dev/sdb";

  boot.initrd.kernelModules = [ "usb_storage" ];

  networking.hostName = "nixos"; # Define your hostname.
  #networking.wireless.enable = true;  # Enables wireless support via 
  networking.networkmanager.enable = true;

  nixpkgs.config.allowUnfree = true;

  #Select internationalisation properties.
  i18n = {
    consoleFont = "Lat2-Terminus16";
    consoleKeyMap = "de";
    defaultLocale = "de_DE.UTF-8";
  };

  # Set your time zone.
  time.timeZone = "Europe/Amsterdam";

  # List packages installed in system profile. To search by name, run:
  # $ nix-env -qaP | grep wget
  environment.systemPackages = with pkgs; [
    pcre
    pkgconfig
    libpqxx
    libmysql
    cabal2nix
    haskellPackages.postgresql-libpq
    haskellPackages.persistent-postgresql
    haskellPackages.zlib
    haskellPackages.yesod
    haskellPackages.yesod-bin
    zlib
    gcc
    oraclejdk8
    stack
    sbt
    scala
    postgresql94
    ghc
    cabal-install
    xorg.xmodmap
    cryptsetup
    wget
    imagemagick
    maven
    youtube-dl
    idea.idea-community
    vlc
    zsh
    unzip
    kde4.okular
    vim
    chromium
    firefox
    thunderbird
    emacs
    networkmanager
    networkmanagerapplet
    gnupg
    git
    lynx
    kde4.yakuake
  ];

  # List services that you want to enable:

  # Enable the OpenSSH daemon.
  # services.openssh.enable = true;

  services.locate.enable = true;
  services.locate.period = "33 7 * * *";
  
  # Enable CUPS to print documents.
  # services.printing.enable = true;

  # Enable the X11 windowing system.
  services.xserver.enable = true;
  services.xserver.layout = "de";
  services.xserver.xkbOptions = "eurosign:e";

  # Enable the KDE Desktop Environment.
  services.xserver.displayManager.kdm.enable = true;
  services.xserver.desktopManager.kde4.enable = true;

  services.postgresql.enable = true;
  services.postgresql.package = pkgs.postgresql94;

  services.mysql = {
    enable = true;
    package = pkgs.mariadb;
  };


  # Define a user account. Don't forget to set a password with ‘passwd’.
  users.extraUsers.guest = {
    isNormalUser = true;
    uid = 1000;
    name = "markus";
    group = "users";
    extraGroups = ["wheel" "networkmanager" "mysql"];
    createHome = true;
    home = "/home/markus";
  };

  # The NixOS release to be compatible with for stateful data such as databases.
  system.stateVersion = "15.09";

}
