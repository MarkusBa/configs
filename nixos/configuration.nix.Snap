# Edit this configuration file to define what should be installed on
# your system.  Help is available in the configuration.nix(5) man page
# and in the NixOS manual (accessible by running ‘nixos-help’).

{ config, pkgs, ... }:

{
  imports =
    [ # Include the results of the hardware scan.
      ./hardware-configuration.nix
    ];

  # Use the GRUB 2 boot loader.
  boot.loader.grub.device = "/dev/sdb";
  boot.loader.grub.enable = true;
  boot.loader.grub.version = 2;
  # Define on which hard drive you want to install Grub.

  networking.hostName = "nixos"; # Define your hostname.
  networking.hostId = "a671cc0d";
  networking.wireless.enable = true;  # Enables wireless.

  # Select internationalisation properties.
  i18n = {
    consoleFont = "lat9w-16";
    consoleKeyMap = "de";
    defaultLocale = "de_DE.UTF-8";
  };

  nixpkgs.config.allowUnfree = true;

  # List packages installed in system profile. To search by name, run:
  # $ nix-env -qaP | grep wget
  environment.systemPackages = with pkgs; [
    #haskellPackages.idris
    #kde4.kdegraphics_thumbnailers
    haskellPackages.ghc
    unzip
    #kde4.ark
    findutils
    haskellPackages.xmonad
    #picolisp
    wpa_supplicant_gui
    vim
    wget
    chromium
    networkmanager
    #network-manager-applet
    emacs
    xlibs.xmodmap
    #xmodmap
    firefox
    thunderbird
    gnupg
    zsh
    #chicken
    lynx
    #mysql
    #mysqlWorkbench
    #oraclejdk8
    pgadmin  
    gitAndTools.gitFull
    postgresql94
    leiningen
    #kde4.yakuake
  ];

  # List services that you want to enable:

  # Enable the OpenSSH daemon.
  # services.openssh.enable = true;

  # Enable CUPS to print documents.
  services.printing.enable = true;

  # Enable the X11 windowing system.
  services.xserver.enable = true;
  services.xserver.layout = "de";
  services.xserver.xkbOptions = "eurosign:e";

  #group networkmanager
  services.xserver.windowManager.icewm.enable = true;
  services.xserver.windowManager.xmonad.enable = true;     # installs xmonad and makes it available
  #services.xserver.windowManager.xmonad.enableContribAndExtras = true; # makes xmonad-contrib and xmonad-extras available
  #services.xserver.windowManager.default       = "xmonad"; # sets it as default
  #services.xserver.desktopManager.default      = "none";   # the plain xmonad experience
  services.xserver.windowManager.default = "icewm";

  # Enable the KDE Desktop Environment.
  services.xserver.displayManager.kdm.enable = true;
  services.xserver.desktopManager.kde4.enable = true;
  
  #services.xserver.displayManager.gdm.enable = true;
  services.xserver.desktopManager.xfce.enable = true;

  # Define a user account. Don't forget to set a password with ‘passwd’.
  users.extraUsers.guest = {
    isNormalUser = true;
    uid = 1000;
    name = "markus";
    group = "users";
    extraGroups = ["wheel" "networkmanager"];
    createHome = true;
    home = "/home/markus";
  };

}
